osprepare/main.go

```diff
@@ -97,31 +159,38 @@ func collectPackages(dist distro, reqs PackageRequirements) []string {
 
 // InstallDeps installs all the dependencies defined in a component
 // specific PackageRequirements in order to enable running the component
-func InstallDeps(reqs PackageRequirements) {
+func InstallDeps(reqs PackageRequirements, logger OSPLog) {
+	if logger == nil {
+		logger = ospNullLogger{}
+	}
 	distro := getDistro()
 
 	if distro == nil {
-		fmt.Fprintf(os.Stderr, "Running on an unsupported distro\n")
+		logger.Errorf("Running on an unsupported distro")
 		if rel := getOSRelease(); rel != nil {
-			fmt.Fprintf(os.Stderr, "Unsupported distro: %s %s\n", rel.Name, rel.Version)
+			logger.Errorf("Unsupported distro: %s %s", rel.Name, rel.Version)
 		} else {
-			fmt.Fprintln(os.Stderr, "No os-release found on this host")
+			logger.Errorf("No os-release found on this host")
 		}
 		return
 	}
+	logger.Infof("OS Detected: %s", distro.getID())
+
 	// Nothing requested to install
 	if reqs == nil {
 		return
 	}
 	if reqPkgs := collectPackages(distro, reqs); reqPkgs != nil {
-		if distro.InstallPackages(reqPkgs) == false {
-			fmt.Fprintf(os.Stderr, "Failed to install: %s\n", strings.Join(reqPkgs, ", "))
+		logger.Infof("Missing packages detected: %v", reqPkgs)
+		if distro.InstallPackages(reqPkgs, logger) == false {
+			logger.Errorf("Failed to install: %s", strings.Join(reqPkgs, ", "))
```

Comment:
@markdryan I'd prefer to just add a patch to this pull request as this change exposes the bug much more clearly.